File: HW 7/161044014_Celal Can_KAYA.pl
male(celalcan).
male(kadir).
male(emre).
male(necati).
male(celal).
male(oktay).
male(kenan).
female(aynur).
female(tuğba).
female(sevgi).
female(emine).
female(müşerref).
female(hanife).
female(latife).
female(bakiye).

parent(müşerref,kadir).
parent(müşerref,kenan).
parent(müşerref,latife).
parent(müşerref,bakiye).
parent(celal,kadir).
parent(celal,kenan).
parent(celal,latife).
parent(celal,bakiye).
parent(hanife,aynur).
parent(hanife,sevgi).
parent(hanife,emine).
parent(hanife,oktay).
parent(necati,aynur).
parent(necati,sevgi).
parent(necati,emine).
parent(necati,oktay).
parent(aynur,celalcan).
parent(aynur,emre).
parent(aynur,tuğba).
parent(kadir,celalcan).
parent(kadir,emre).
parent(kadir,tuğba).

child(kadir,müşerref).
child(kadir,celal).
child(kenan,müşerref).
child(kenan,celal).
child(latife,müşerref).
child(latife,celal).
child(bakiye,müşerref).
child(bakiye,celal).
child(aynur,hanife).
child(aynur,necati).
child(sevgi,hanife).
child(sevgi,necati).
child(emine,hanife).
child(emine,necati).
child(oktay,hanife).
child(oktay,necati).
child(celalcan,aynur).
child(celalcan,kadir).
child(emre,aynur).
child(emre,kadir).
child(tuğba,aynur).
child(tuğba,kadir).

sibling(kadir,kenan).
sibling(kenan,kadir).
sibling(kadir,latife).
sibling(latife,kadir).
sibling(kadir,bakiye).
sibling(bakiye,kadir).
sibling(kenan,latife).
sibling(latife,kenan).
sibling(kenan,bakiye).
sibling(bakiye,kenan).
sibling(latife,bakiye).
sibling(bakiye,latife).
sibling(aynur,sevgi).
sibling(sevgi,aynur).
sibling(aynur,emine).
sibling(emine,aynur).
sibling(aynur,oktay).
sibling(oktay,aynur).
sibling(sevgi,emine).
sibling(emine,sevgi).
sibling(sevgi,oktay).
sibling(oktay,sevgi).
sibling(emine,oktay).
sibling(oktay,emine).
sibling(celalcan,tuğba).
sibling(tuğba,celalcan).
sibling(celalcan,emre).
sibling(emre,celalcan).
sibling(emre,tuğba).
sibling(tuğba,emre).

mother(X,Y) :- female(X),parent(X,Y).
father(X,Y) :- male(X),parent(X,Y).
brother(X,Y) :- male(X),sibling(X,Y).
sister(X,Y) :- female(X),sibling(X,Y).
daughter(X,Y) :- female(X),child(X,Y).
son(X,Y) :- male(X),child(X,Y).
grandmother(X,Y) :- child(Y,Z), child(Z,X).
aunt(X,Y) :- female(X),sibling(X,Z),parent(Z,Y).

File: HW 7/161044014_Celal Can_KAYA.py
def member():
    members = ['celal', 'emre', 'tuğba', 'aynur', 'kadir', 'sevgi', 'oktay', 'emine', 'hanife', 'necati']
    print("Members of the family: " + str(members) + "\nWrite the 2 members of this family for see their relationship.\n" )

def get_input_1():
    x=input("First:")

    return x

def get_input_2():
    y=input("Second Member:")

    return y


def relation():
    relations = {"celal emre" : "brother", "celal aynur": "son", "celal tuğba": "brother", "celal kadir": "son", "celal emine": "nephew", "celal sevgi": "nephew",
                 "celal oktay": "nephew", "celal necati": "grandson", "celal hanife": "grandson", "emre celal": "brother", "emre tuğba": "brother",
                 "emre aynur": "son", "emre kadir": "son", "emre sevgi": "nephew", "emre emine": "nephew", "emre oktay": "nephew", "emre necati": "grandson",
                 "emre hanife": "grandson", "tuğba celal": "sister", "tuğba emre": "sister", "tuğba aynur": "daughter", "tuğba kadir": "daughter",
                 "tuğba sevgi": "nephew", "tuğba emine": "nephew", "tuğba oktay" : "nephew", "tuğba necati" : "granddaughter", "tuğba hanife": "granddaughter",
                 "aynur celal": "mother", "aynur emre": "mother", "aynur tuğba": "mother", "aynur kadir": "wife", "aynur sevgi": "sister", "aynur emine":"sister",
                 "aynur oktay": "sister", "aynur hanife": "daughter", "aynur necati": "daughter", "kadir celal": "father", "kadir tuğba": "father",
                 "kadir emre": "father", "kadir aynur": "husband", "kadir sevgi": "brother in law", "kadir emine": "brother in law", "kadir oktay": "brother in law",
                 "kadir necati": "son in law", "kadir hanife": "son in law", "sevgi celal": "aunt", "sevgi tuğba": "aunt", "sevgi emre": "aunt",
                 "sevgi kadir": "sister in law", "sevgi emine": "sister", "sevgi aynur": "sister", "sevgi oktay": "sister", "sevgi hanife": "daughter",
                 "sevgi necati": "daughter", "emine celal": "aunt", "emine tuğba": "aunt", "emine emre": "aunt", "emine kadir": "sister in law",
                 "emine aynur": "sister", "emine sevgi": "sister", "emine oktay": "sister", "emine hanife": "daughter", "emine necati": "daughter",
                 "oktay celal": "uncle", "oktay emre": "uncle", "oktay tuğba": "uncle", "oktay kadir": "brother in law", "oktay aynur": "brother",
                 "oktay sevgi": "brother", "oktay emine": "brother", "oktay hanife": "son", "oktay necati": "son", "hanife celal": "grandmother",
                 "hanife emre": "grandmother", "hanife tuğba": "grandmother", "hanife aynur": "mother", "hanife kadir": "mother in law", "hanife sevgi": "mother",
                 "hanife oktay": "mother", "hanife emine": "mother", "hanife necati": "wife", "necati celal": "grandfather", "necati emre": "grandfather",
                 "necati tuğba": "grandfather", "necati aynur": "father", "necati kadir": "father in law", "necati sevgi": "father", "necati emine": "father",
                 "necati oktay": "father", "necati hanife": "husband"}
                 
                 

    return relations

def main():
    member()
    x=get_input_1()
    y=get_input_2()
    relations=relation()

    z = str(x) + " " + str(y)
    v = str(y) + " " + str(x)

    if z in relations:
        print("\n" + x + " is " + relations[z] + " of " + y + ".")
        if v in relations:
            print(y + " is " + relations[v] + " of " + x + ".")
    else:
        print("Not Found!")

main()

